parsl/monitoring/app_monitor.py

```diff
@@ -5,59 +5,87 @@
             try:
-                d['psutil_process_disk_write'] += to_mb(child.io_counters().write_bytes)
-                d['psutil_process_disk_read'] += to_mb(child.io_counters().read_bytes)
+                d['psutil_process_disk_write'] = to_mb(pm.io_counters().write_bytes)
+                d['psutil_process_disk_read'] = to_mb(pm.io_counters().read_bytes)
             except psutil._exceptions.AccessDenied:
-                pass
-        logger.info("test", extra=d)
-        time.sleep(5)
+                # occassionally pid temp files that hold this information are unvailable to be read so set to zero
+                d['psutil_process_disk_write'] = 0
+                d['psutil_process_disk_read'] = 0
+            for child in children:
+                for k, v in child.as_dict(attrs=summable_values).items():
+                    d['psutil_process_' + str(k)] += v
+                d['psutil_process_time_user'] += child.cpu_times().user
+                d['psutil_process_time_system'] += child.cpu_times().system
+                d['psutil_process_memory_virtual'] += to_mb(child.memory_info().vms)
+                d['psutil_process_memory_resident'] += to_mb(child.memory_info().rss)
+                try:
+                    d['psutil_process_disk_write'] += to_mb(child.io_counters().write_bytes)
+                    d['psutil_process_disk_read'] += to_mb(child.io_counters().read_bytes)
+                except psutil._exceptions.AccessDenied:
+                    # occassionally pid temp files that hold this information are unvailable to be read so add zero
+                    d['psutil_process_disk_write'] += 0
+                    d['psutil_process_disk_read'] += 0
+
+            to_be_rounded = ['psutil_process_memory_virtual', 'psutil_process_memory_resident', 'psutil_process_cpu_percent', 'psutil_process_memory_percent',
+                             'psutil_process_disk_write', 'psutil_process_disk_read']
+            for k in to_be_rounded:
+                d[k] = round(d[k], 2)
+
+        finally:
+            logger.info("task resource update", extra=d)
+            sleep_duration = db_logger_config_object.resource_loop_sleep_duration
+            time.sleep(sleep_duration)
 
 
-def monitor_wrapper(f, task_id, db_logger_config, run_id):
+def monitor_wrapper(f, task_id, db_logger_config_object, run_id):
+    """ Internal
+    Wrap the Parsl app with a function that will call the monitor function and point it at the correct pid when the task begins.
+    """
     def wrapped(*args, **kwargs):
-        p = Process(target=monitor, args=(os.getpid(), task_id, db_logger_config, run_id))
+        p = Process(target=monitor, args=(os.getpid(), task_id, db_logger_config_object, run_id))
         p.start()
         try:
             return f(*args, **kwargs)
         finally:
+            # Note: I believe finally blocks are not called if the python parsl process is killed with certain SIGs which may leave the monitor as a zombie
```

Comment:
If the parsl process is killed with `SIGKILL`, then no one gets any chance to shut down nicely (including leaving all kinds of stuff running, not just this)

Other signals (SIGTERM, SIGINT) I think will run this finaliser (or at least, if not, other parts of parsl also have problems...)